### PR TITLE
chore(performance): create initial Makefile [INFENG-223]

### PR DESCRIPTION
**Pull request by loksonarius**
## Test Plan

Changes tested locally. Future work will test the configuration code in CI runners.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: performance/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Docker image
+IMAGE_REPO ?= determinedai/perf-test
+IMAGE_TAG ?= $(USER)
+
+# Perf testing cluster
+CLUSTER_ID ?= $(USER)-perf
+KEYPAIR_ID ?= $(USER)
```
Comment (stoksc) [inline performance/Makefile:7]: idk if this is typically true, so maybe add a README to explain how to override it.

File: performance/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	cat .up.out \
+		| grep DET_MASTER \
+		| sed -rn 's/.*DET_MASTER=(.*)/\1/p' \
+		> .det.url
+
+down:
+	det deploy aws down --cluster-id $(CLUSTER_ID)
+
+run:
+	docker run $(IMAGE) \
```
Comment (stoksc) [inline performance/Makefile:38]: QoL: check if DET_URL == UNSET and bail? or `run: up.stamp` or something.

File: performance/Makefile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+VERSION ?= $(shell det -v | cut -d' ' -f2)
+
+# Perf test options
+DET_ADMIN_USERNAME ?= admin
+DET_ADMIN_PASSWORD ?= ""
+
+IMAGE = $(IMAGE_REPO):$(IMAGE_TAG)
+DET_URL ?= $(shell cat .det.url || echo UNSET)
+
+build: Dockerfile src/* package-lock.json package.json tsconfig.json webpack.config.js
```
Comment (stoksc) [inline performance/Makefile:17]: should the default target be something else?

File: performance/Makefile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	| tee .up.out && \
+	cat .up.out \
+		| grep DET_MASTER \
+		| sed -rn 's/.*DET_MASTER=(.*)/\1/p' \
+		> .det.url
+
+down:
+	det deploy aws down --cluster-id $(CLUSTER_ID)
+
+run:
```
Comment (stoksc) [inline performance/Makefile:37]: my be nice to set ourselves up for lowest CI costs by adding a `run_local` target and recommending folks iterate on test writing with `run-local` and graduate to `run` when they need to actually test performance.

**Review by stoksc — APPROVED** [review]
lgtm. I could take or leave most of my comments.

File: performance/Makefile (hunk 3 above)
Comment (loksonarius) [reply performance/Makefile:17]: Is there a specific target you feel makes most sense?

File: performance/Makefile (hunk 4 above)
Comment (loksonarius) [reply performance/Makefile:37]: Yeah that's a good point.
I'll try implementing it real quick using `det deploy local` and see how it goes.
I'll mention running `-local` targets in a README.

File: performance/Makefile (hunk 1 above)
Comment (loksonarius) [reply performance/Makefile:7]: It's not meant to be necessarily true, but more so something that'll break intentionally -- we don't have any real mandate on key pair names for individual devs, so not much to do here with it.
I can add an initial README that lists the overridable Makefile vars for now

Comment (stoksc) [reply performance/Makefile:7]: I think a good readme for this and how to config your make calls will probably save time in the long run

**Review by stoksc — APPROVED** [review]

**Pull request merged**